clients/drcachesim/reader/zipfile_file_reader.cpp

```diff
@@ -87,28 +87,31 @@ file_reader_t<zipfile_reader_t>::read_next_entry()
         if (num_read == 0) {
 #ifdef DEBUG
             if (verbosity_ >= 3) {
-                char name[128];
-                name[0] = '\0'; /* Just in case. */
+                zipfile->name[0] = '\0'; /* Just in case. */
                 // This call is expensive if we do it every time.
-                unzGetCurrentFileInfo64(zipfile->file, nullptr, name, sizeof(name),
-                                        nullptr, 0, nullptr, 0);
-                VPRINT(this, 3, "Hit end of component %s; opening next component\n",
-                       name);
+                unzGetCurrentFileInfo64(zipfile->file, nullptr, zipfile->name,
+                                        sizeof(zipfile->name), nullptr, 0, nullptr, 0);
+                VPRINT(this, 3, "Hit end of component %s; opening next component in %s\n",
+                       zipfile->name, zipfile->path.c_str());
             }
 #endif
             // read_next_entry() stored the last-read entry into entry_copy_.
             if ((entry_copy_.type != TRACE_TYPE_MARKER ||
                  entry_copy_.size != TRACE_MARKER_TYPE_CHUNK_FOOTER) &&
                 entry_copy_.type != TRACE_TYPE_FOOTER) {
-                VPRINT(this, 1, "Chunk is missing footer: truncation detected\n");
+                zipfile->name[0] = '\0'; /* Just in case. */
+                unzGetCurrentFileInfo64(zipfile->file, nullptr, zipfile->name,
```

Comment:
This is only to get the filename? Isn't that in input_file_->path now?

Reply:
That new path field is indeed the filename, but here we want the component name inside the zipfile (for ours, "chunk.0006" e.g.).